relay-server/src/actors/global_config.rs

```diff
@@ -271,3 +279,37 @@ impl Service for GlobalConfigService {
         });
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use relay_system::{Controller, Service, ShutdownMode};
+    use relay_test::mock_service;
+    use std::sync::Arc;
+    use std::time::Duration;
+
+    use crate::actors::global_config::{Get, GlobalConfigService};
+
+    /// Tests that the service can still handle requests after sending a
+    /// shutdown signal.
+    #[tokio::test]
+    async fn test_service_shutdown() {
```

Comment:
This test does not ensure the service doesn't make upstream requests.
I'm not sure if it's worth the time and effort to test this -- the global
config service doesn't support generics for `upstream` and the service
mocking is rather naive.